.github/workflows/hil_tests.yml

```diff
@@ -407,7 +410,17 @@ jobs:
             $hil_tracefiles $twister_tracefiles \
             -e examples -e external -e tests"
 
-          gcovr --txt-summary --xml coverage.xml
+          rm -rf coverage
+          mkdir -p coverage
+          gcovr --txt-summary --xml coverage.xml --html-details coverage/index.html
+
+      - name: Upload coverage artifacts
+        uses: actions/upload-artifact@v4
```

Comment:
Can we use safe-upload-artifacts here?

Reply:
Sure we can... but should we? There are just HTML generated pages, with code and colors applied based on coverage. There are no logs there, so no secrets. IMO we should use generic stuff for things that do not require special treatment.

Reply:
I'm also hesitant on this one in particular because we could (I don't think likely, but possible) inadvertently change the source code that's displayed in the HTML report which could make searching or comparing more difficult.